samza-core/src/main/java/org/apache/samza/processor/StreamProcessor.java

```diff
@@ -179,27 +182,36 @@ public synchronized void stop() {
     boolean containerShutdownInvoked = false;
     if (container != null) {
       try {
-        LOGGER.info("Shutting down container " + container.toString() + " from StreamProcessor");
+        LOGGER.info("Shutting down the container: {} of stream processor: {}.", container, processorId);
         container.shutdown();
+        LOGGER.info("Waiting {} milliseconds for the container: {} to shutdown.", taskShutdownMs, container);
+        containerShutdownLatch.await(taskShutdownMs, TimeUnit.MILLISECONDS);
+        if (executorService != null) {
+          executorService.shutdown();
+          if (!executorService.isShutdown()) {
```

Comment:
Issue: This will always be false, since isShutdown returns whether shutdown or shutdownNow has been invoked by the user, not whether any threads are currently executing.

Reply:
The right method to invoke there is isTerminated, which returns true if all the tasks in that `executorService` had completed. However we have to invoke shutdown or shutdownNow before that(from isTerminated javadoc).
Switched to that.

Reply:
Switched to use isTerminated method, which returns true if all the tasks in a `executorService` had completed.

However we should invoke  executorService.shutdown or executorService.shutdownNow prior to that(from isTerminated javadoc).

When the initial container shutdown was successful, `isTerminated` check prevents an additional `task.shutdown.ms` wait.